crypto/evp/pkey_mac.c

```diff
@@ -441,6 +441,24 @@ static int pkey_mac_ctrl_str(EVP_PKEY_CTX *ctx,
     OSSL_PARAM params[2];
     int ok = 0;
 
+    /*
+     * Translation of some control names that are equivalent to a single
+     * parameter name.
+     *
+     * md, digest and cipher are really just the underlying algorithm,
+     * and since they are now simply specified by name, we use the single
+     * parameter "algorithm" for them all.
+     *
+     * "digestsize" was a setting control in siphash, but naming wise,
+     * it's really the same as "size".
+     */
+    if (strcmp(type, "md") == 0
```

Comment:
Are you sure about this.. 
In kdf at least there can be two algorithms so this wont work for that case.. I dont know if I like this.

Reply:
I guess what I am saying is that it is going to be inconsistent with KDF.

Reply:
The only I can find is sskdf, which can have a MAC and an MD.  So it may make sense to have "digest", "cipher" and "mac" for KDFs...  and yeah, in the interest of not having too much diversity, it might be suitable to have "digest" and "cipher" for MACs as well, then.

Do you agree, though, that "md" and "digest" can be merged into one, i.e. one gets translated into the other (I would rather standardise on "digest")?

Reply:
Yes I agree..

Reply:
Cool, that's a good enough compromise for me :-)